File: test/single-language/csharp/dotnetsdk/http-service/success-deployment-smoke/Services/SummaryService.cs
using System.Text.RegularExpressions;
namespace Fixture;
public sealed class SummaryService
{
    public Summary Summarize(string? raw)
    {
        var tokens = (raw ?? "1,2,3").Split(',');
        if (tokens.Length > 20) throw new ArgumentException("invalid-values");
        var items = new List<int>();
        foreach (var token in tokens)
        {
            if (!Regex.IsMatch(token, "^[0-9]{1,10}$") || !long.TryParse(token, out var value) || value < 0 || value > 10000)
                throw new ArgumentException("invalid-values");
            items.Add((int)value);
        }
        return new Summary(items);
    }
}

File: test/single-language/ruby/bundler/http-service/success-deployment-smoke/lib/summary_service.rb
require_relative 'summary'
module Fixture
  class SummaryService
    def summarize(raw)
      tokens = raw.nil? ? ['1', '2', '3'] : raw.split(',', -1)
      raise ArgumentError, 'invalid-values' unless (1..20).cover?(tokens.length)

      items = tokens.map do |token|
        raise ArgumentError, 'invalid-values' unless token.match?(/\A[0-9]{1,10}\z/) && token.to_i <= 10000

        token.to_i
      end
      Summary.new(items)
    end
  end
end
